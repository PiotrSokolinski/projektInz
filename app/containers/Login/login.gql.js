import gql from 'graphql-tag'

export default gql`
  mutation loginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      id
      email
      firstName
      lastName
      role
      avatarUrl
      token
      color
      group {
        id
      }
    }
  }
`
