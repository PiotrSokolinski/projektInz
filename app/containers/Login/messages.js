/*
 * Login Messages
 *
 * This contains all the text for the Login container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.Login'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Sing in',
  },
  buttonTitle: {
    id: `${scope}.buttonTitle`,
    defaultMessage: 'Login',
  },
  emailLabel: {
    id: `${scope}.emailLabel`,
    defaultMessage: 'E-mail address',
  },
  passwordLabel: {
    id: `${scope}.passwordLabel`,
    defaultMessage: 'Password',
  },
  emailInvalidError: {
    id: `${scope}.emailInvalidError`,
    defaultMessage: 'E-mail address is invalid',
  },
  emailEmptyError: {
    id: `${scope}.emailEmptyError`,
    defaultMessage: 'E-mail address is required',
  },
  passwordEmptyError: {
    id: `${scope}.passwordEmptyError`,
    defaultMessage: 'Password is required',
  },
  forgotPasswordMessage: {
    id: `${scope}.forgotPasswordMessage`,
    defaultMessage: 'Lost your password?',
  },
  errorTitle: {
    id: `${scope}.errorTitle`,
    defaultMessage: 'Something went wrong',
  },
  noAccount: {
    id: `${scope}.noAccount`,
    defaultMessage: `Don't have an account?`,
  },
})
