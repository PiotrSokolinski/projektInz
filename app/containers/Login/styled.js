import styled from 'styled-components'
import { Link as RouterLink } from 'react-router-dom'
import { Help } from 'react-bulma-components/lib/components/form'
import { Lock } from 'styled-icons/material'
import { User } from 'styled-icons/boxicons-solid'
import { Colors } from 'themes'

export const Error = styled(Help)`
  top: -15px;
  position: relative;
`

export const Box = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 400px;
  padding: 40px;
  background: rgba(0, 0, 0, 0.8);
  box-sizing: border-box;
  box-shadow: 0 15px 25px rgba(0, 0, 0, 0.5);
  border-radius: 10px;
`

export const Header = styled.h1`
  margin: 0 0 30px;
  padding: 0;
  color: #fff;
  text-align: center;
`
export const InputBox = styled.div`
  position: relative;
`
export const Input = styled.input`
  width: 100%;
  padding: 10px 0;
  font-size: 16px;
  color: #fff;
  letter-spacing: 1px;
  margin-bottom: 30px;
  border: none;
  outline: none;
  background: transparent;
  border-bottom: 1px solid #fff;
  text-indent: 30px;
`

export const Label = styled.label`
  position: absolute;
  top: 22px;
  left: 30px;
  padding: 10px 0;
  font-size: 16px;
  color: white;
  pointer-events: none;
  transition: 0.5s;
  ${props => (props.isEmpty ? 'top: -15px; left: 0; color: #03a9f4;' : '')};
`

export const Button = styled.button`
  background: transparent;
  border: none;
  outline: none;
  color: #fff;
  background: ${Colors.darkBlue};
  padding: 10px 20px;
  cursor: pointer;
  border-radius: 5px;
  transition: 0.5s;
  width: 150px;
  &:hover {
    transform: scale(1.1);
    filter: brightness(75%);
  }
`
const icon = `
  width: 25px;
  float: left;
  text-align: center;
  color: ${Colors.darkBlue};
  top: 30px;
  position: relative;
`

export const UserIcon = styled(User)`
  ${icon}
`
export const LockIcon = styled(Lock)`
  ${icon}
`
export const Avatar = styled.img`
  position: absolute;
  height: 100px;
  width: 100px;
  border-radius: 50%;
  left: calc(50% - 50px);
  top: -50px;
`
export const Link = styled(RouterLink)`
  float: right;
  text-decoration: none;
  color: white;
  font-size: 12px;
  line-height: 20px;
  transition: 0.5s;
  &:hover {
    color: #03a9f4;
    transform: scale(1.1);
  }
`

export const Wrapper = styled.div`
  & ${Input}:hover + ${Label} {
    top: -15px;
    left: 0;
    color: #03a9f4;
  }
`
