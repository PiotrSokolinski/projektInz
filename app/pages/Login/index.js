import React from 'react'
import LoginContainer from 'containers/Login'
const Login = props => <LoginContainer {...props} />

export default Login
