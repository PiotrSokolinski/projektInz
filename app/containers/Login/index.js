/**
 *
 * Login
 *
 */

import React from 'react'
import * as Yup from 'yup'
import PropTypes from 'prop-types'
import { FormattedMessage, injectIntl } from 'react-intl'
import { Formik } from 'formik'

import Button from 'components/Button'
import PublicInput from 'components/PublicInput'

import messages from './messages'
import * as Styled from './styled'

const initialValues = { email: '', password: '' }

const validationSchema = intl =>
  Yup.object().shape({
    email: Yup.string()
      .email(intl.formatMessage(messages.emailInvalidError))
      .required(intl.formatMessage(messages.emailEmptyError)),
    password: Yup.string().required(intl.formatMessage(messages.passwordEmptyError)),
  })

const Login = ({ intl, history }) => {
  const submitLoginForm = () => {
    history.push('/')
  }
  return (
    <Styled.Container>
      <Styled.Box>
        <Styled.HeaderContainer>
          <Styled.Avatar size="150" />
          <Styled.Header>
            <FormattedMessage {...messages.header} />
          </Styled.Header>
        </Styled.HeaderContainer>
        <Formik onSubmit={submitLoginForm} initialValues={initialValues} validationSchema={validationSchema(intl)}>
          {({ errors, touched, values, handleChange, handleBlur, handleSubmit }) => (
            <form autoComplete="off">
              <PublicInput
                id="emailField"
                icon={Styled.UserIcon}
                label={intl.formatMessage(messages.emailLabel)}
                error={touched.email && errors.email}
                inputProps={{
                  type: 'email',
                  name: 'email',
                  value: values.email,
                  onBlur: handleBlur,
                  onChange: handleChange,
                }}
              />
              <PublicInput
                id="passwordField"
                icon={Styled.LockIcon}
                label={intl.formatMessage(messages.passwordLabel)}
                error={touched.password && errors.password}
                inputProps={{
                  type: 'password',
                  name: 'password',
                  value: values.password,
                  onBlur: handleBlur,
                  onChange: handleChange,
                }}
              />
              <Styled.FooterContainer>
                <Button type="submit" onClick={handleSubmit}>
                  <FormattedMessage {...messages.buttonTitle} />
                </Button>
                <Styled.LinksContainer>
                  <Styled.Link to="/password-remind">
                    <FormattedMessage {...messages.forgotPasswordMessage} />
                  </Styled.Link>
                  <Styled.Link to="/registration">
                    <FormattedMessage {...messages.noAccount} />
                  </Styled.Link>
                </Styled.LinksContainer>
              </Styled.FooterContainer>
            </form>
          )}
        </Formik>
      </Styled.Box>
    </Styled.Container>
  )
}

Login.propTypes = {
  intl: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
}

export default injectIntl(Login)
