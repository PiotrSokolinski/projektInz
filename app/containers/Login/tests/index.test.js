/**
 *
 * Tests for Login
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import { MemoryRouter } from 'react-router-dom'

import Login from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<Login />', () => {
  const history = {
    push: jest.fn(),
  }
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <Login history={history} />
        </MemoryRouter>
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <Login history={history} />
        </MemoryRouter>
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
