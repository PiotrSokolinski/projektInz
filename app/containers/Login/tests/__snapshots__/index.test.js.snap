// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Should render and match the snapshot 1`] = `
<div
  class="styled__Box-al6wyn-1 bBTkVe"
>
  <img
    alt="loginIcon"
    class="styled__Avatar-al6wyn-9 zaPyc"
    src="IMAGE_MOCK"
  />
  <h1
    class="styled__Header-al6wyn-2 dXIdJm"
  >
    <span>
      Sing in
    </span>
  </h1>
  <form
    autocomplete="off"
  >
    <div
      class="styled__InputBox-al6wyn-3 juArMf"
    >
      <svg
        aria-hidden="true"
        class="styled__UserIcon-al6wyn-7 ggcOXx StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.5 6.5C7.5 8.981 9.519 11 12 11s4.5-2.019 4.5-4.5S14.481 2 12 2 7.5 4.019 7.5 6.5zM20 21h1v-1c0-3.859-3.141-7-7-7h-4c-3.86 0-7 3.141-7 7v1h17z"
        />
      </svg>
      <div
        class="styled__Wrapper-al6wyn-11 gbYXzT"
      >
        <input
          class="styled__Input-al6wyn-4 ecYVYH"
          name="email"
          type="email"
          value=""
        />
        <label
          class="styled__Label-al6wyn-5 koFebL"
        >
          E-mail address
        </label>
      </div>
    </div>
    <div
      class="styled__InputBox-al6wyn-3 juArMf"
    >
      <svg
        aria-hidden="true"
        class="styled__LockIcon-al6wyn-8 keBdnB StyledIconBase-sc-bdy9j4 jbPeIR"
        fill="currentColor"
        focusable="false"
        height="22"
        viewBox="0 0 24 24"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
        />
      </svg>
      <div
        class="styled__Wrapper-al6wyn-11 gbYXzT"
      >
        <input
          class="styled__Input-al6wyn-4 ecYVYH"
          name="password"
          type="password"
          value=""
        />
        <label
          class="styled__Label-al6wyn-5 koFebL"
        >
          Password
        </label>
      </div>
    </div>
    <button
      class="styled__Button-al6wyn-6 dpAknp"
      type="submit"
    >
      <span>
        Login
      </span>
    </button>
    <a
      class="styled__Link-al6wyn-10 dhqPmV"
      href="/password-remind"
    >
      <span>
        Lost your password?
      </span>
    </a>
    <a
      class="styled__Link-al6wyn-10 dhqPmV"
      href="/register"
    >
      <span>
        Don't have an account?
      </span>
    </a>
  </form>
</div>
`;
